### ISSUE TITLE
Is the dropout documentation correct?

### ISSUE THREAD

**Issue by ahmedhosny** [keras-team/keras#7979]
**Not related to the various methods one can go about applying dropout.**
**Related to cross referencing keras dropout documentation & tensorflow_backend.py**

```
from keras import backend as K
x = np.array([1,2,3,4,5,6])
x = K.variable(value = x, dtype = 'float64')
out = K.dropout(x, 0.5)
K.eval(out)
```
possible outcomes:
```
array([ 0.,  4.,  0.,  0.,  0.,  0.])
array([  0.,   4.,   6.,   8.,  10.,   0.])
array([ 2.,  4.,  6.,  0.,  0.,  0.])
```

When running the above code, it is doing exactly what is defined [here](https://github.com/tensorflow/tensorflow/blob/r1.3/tensorflow/python/ops/nn_ops.py#L1864) in the tf source: 
1. scaling up all by 1/rate
2. then setting a random fraction to zero, stress here on the word random i.e. not user-defined

The keras [docs](https://keras.io/layers/core/#dropout) and [source](https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L72) however, both state that "Dropout consists in randomly setting a fraction rate of input units to 0 at each update during training time, which helps prevent overfitting." with the user-defined rate argument being: "float between 0 and 1. Fraction of the input units to drop."

any ideas or did I miss something?
I ask only because of the keras [tensorflow_backend.py](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2867) stating that dropout "Sets entries in `x` to zero at random, while scaling the entire tensor.", which makes sense with the experiments and the tf source, but not with the keras docs.
